### ISSUE TITLE
Specifying "--dns-opt ndots:X" should override the default "ndots:0"

### ISSUE THREAD

**Issue by fermayo** [docker/docker#26039]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 1
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 3
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host overlay bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.954 GiB
Name: moby
ID: UI36:ZFB5:JVVI:6TR5:YI4Z:7ZAG:DI5M:UBOM:DCJW:OFXX:O4E5:BP5U
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 29
 System Time: 2016-08-26T09:54:35.208231913Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Username: fermayo
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Using Docker for Mac

**Steps to reproduce the issue:**
1. `docker network create test`
2. `docker run --net test --dns-opt ndots:1 alpine cat /etc/resolv.conf`

**Describe the results you received:**

```
search local
nameserver 127.0.0.11
options ndots:1 ndots:0
```

**Describe the results you expected:**

```
search local
nameserver 127.0.0.11
options ndots:1
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by justincormack** [comment]
There are actually tests for this but they string match and pass even with the extra `ndots:0` on the end! https://github.com/docker/docker/blob/master/integration-cli/docker_cli_run_test.go#L1299

**Comment by justincormack** [comment]
The code is actually in libnetwork, so maybe open an issue there too linking here?

cc @mrjana

**Comment by mghazizadeh** [comment]
Does this issue exist in 1.11 as well or is this a 1.12 regression? If it's in 1.11 where would it be fixed?

**Comment by sanimej** [comment]
This is not a regression. It has been the behavior since 1.10 where we introduced embedded DNS server. Overriding ndots:0 will prevent resolution of unqualified names by the docker DNS server if the resolv.conf file has search domain option. Fix for this will be in libnetwork and we can make it available for the next 1.12 patch release.

**Comment by mghazizadeh** [comment]
Paying cloud customers have this issue in 1.11 and cloud follows the engine CS model, where only Docker CS version is the supported in run time engine. Is there a plan to patch this in 1.11 CS?
